db/migrations/20190919141224_create_ilk_state_history_table_and_triggers.sql

```diff
@@ -247,23 +293,28 @@ $$
 -- +goose StatementEnd
 
 -- +goose StatementBegin
-CREATE OR REPLACE FUNCTION maker.insert_ilk_rates() RETURNS TRIGGER
+CREATE OR REPLACE FUNCTION maker.update_ilk_rates() RETURNS TRIGGER
 AS
 $$
 BEGIN
-    PERFORM maker.insert_new_rate(NEW);
-    PERFORM maker.update_later_rates(NEW);
+    IF (TG_OP IN ('INSERT', 'UPDATE')) THEN
+        PERFORM maker.insert_new_rate(NEW);
+        PERFORM maker.update_rates_until_next_diff(NEW, NEW.rate);
+    ELSIF (TG_OP = 'DELETE') THEN
+        PERFORM maker.update_rates_until_next_diff(OLD, ilk_rate_before_block(OLD.ilk_id, OLD.block_number));
+        PERFORM maker.delete_redundant_ilk_state(OLD.ilk_id, OLD.block_number);
+    END IF;
```

Comment:
Not sure what the best practices are for triggers (i.e. whether it's better to have multiple triggers on the same table or use conditional logic like this), but this seemed preferable to me. Happy to change if we want to.